api/scpca_portal/models/job.py

```diff
@@ -104,6 +111,27 @@ def get_sample_job(
             },
         )
 
+    @classmethod
+    def get_retry_jobs(cls) -> List[Self] | None:
+        """
+        Prepare new unsaved Job instances to retry terminated jobs.
+        Exclude those with critical_error is True and retry_on_termination is False.
+        Set each instance's attempt to the base terminated job's attempt incremented by 1.
+        Return the new instances, otherwise None if no terminated jobs for retry available.
+        """
+        if terminated_jobs := Job.objects.filter(
+            state=JobStates.TERMINATED.name, retry_on_termination=True
+        ).exclude(critical_error=True):
+            total_retry_job_count = len(terminated_jobs)
+            logger.info(
+                f"{total_retry_job_count} job{pluralize(total_retry_job_count)} "
+                "generated for retrying terminated jobs.",
+            )
+            return [terminated_job.get_retry_job() for terminated_job in terminated_jobs]
```

Comment:
Currently, I am returning a list of generated instances instead of bulk saving them, as they are still in the `CREATED` state (before job submission). Let me know your thoughts on this. Thank you, David!